### DISCUSSION TITLE
How to initialize the status of NavigationView to select the first one by default?

### DISCUSSION THREAD

**Discussion by sxlllslgh** [microsoft/WindowsAppSDK#3088] [Q&A]
I write an simple demo of NavigationView, the XAML is:
```xaml
<NavigationView x:Name="mainNavigation" MenuItemsSource="{x:Bind MainWindowNavigationCategoryViewModel.NavigationCategories, Mode=OneWay}" Loaded="mainNavigationLoaded" ItemInvoked="mainNavigationItemInvoked">
    <NavigationView.MenuItemTemplate>
        <DataTemplate x:DataType="local:NavigationCategory">
            <NavigationViewItem Content="{x:Bind Name}">
                <NavigationViewItem.Icon>
                    <SymbolIcon Symbol="{x:Bind Icon}"/>
                </NavigationViewItem.Icon>
            </NavigationViewItem>
        </DataTemplate>
    </NavigationView.MenuItemTemplate>
    <Frame x:Name="contentFrame"/>
</NavigationView>
```
The `MenuItemsSource` is a `Windows::Foundation::Collections::IObservableVector` implemented by `winrt::single_threaded_observable_vector`. I add items in the construtor:

```cpp
MainWindow::MainWindow() {
    InitializeComponent();
    mainWindowNavigationCategoryViewModel.NavigationCategories().Append(make<MyApp::implementation::NavigationCategory>(/*Parameters of item*/));
}
```

Now I want to let the first `NavigationViewItem` be selected once the window opened, so I wirte a function bound to `Loaded` event:
```cpp
void winrt::MyApp::implementation::MainWindow::mainNavigationLoaded(winrt::Windows::Foundation::IInspectable const& sender, winrt::Microsoft::UI::Xaml::RoutedEventArgs const& e) {
    mainNavigation().SelectedItem(mainNavigation().MenuItems().First());
}
```
I also tried:
```cpp
void winrt::MyApp::implementation::MainWindow::mainNavigationLoaded(winrt::Windows::Foundation::IInspectable const& sender, winrt::Microsoft::UI::Xaml::RoutedEventArgs const& e) {
    mainNavigation().SelectedItem(mainNavigation().MenuItemsSource().as<Windows::Foundation::Collections::IObservableVector<MyApp::NavigationCategory>>().First());
}
```
But both methods not work. How to solve this problem? Thanks.

**Comment by PMoffitt05** [comment]
For C#, In MainNavigationLoaded this selects the 0th item from the list so something is selected:
        MainNavigation.SelectedItem = MainNavigation.MenuItems[0];

However, the MenuItems need to be defined prior to that code otherwise I would assume it is setting the SelectedItem to null.

Pretty much pledgurized that from the Windowing gallery sample.
I am not sure how that relates to your code but it may be some help as that looks like C++ code.  There is also a C++ code version of the Windowing gallery sample (https://github.com/microsoft/WindowsAppSDK-Samples.git).
